formspree/app.py

```diff
@@ -2,18 +2,31 @@
 from flask import g
 from flask.ext.sqlalchemy import SQLAlchemy
 from flask.ext.login import LoginManager, current_user
-import fakeredis, redis
+import redis
 import settings
 
+
 DB = SQLAlchemy()
-if settings.TESTING:
-    REDIS = fakeredis.FakeStrictRedis()
-else:
-    REDIS = redis.StrictRedis.from_url(settings.REDIS_URL)
+_redis = None
+
+
+def REDIS():
+    ''' function to acquire the global redis connection '''
+    global _redis
+    if not _redis:
+        _redis = get_redis()
+    return _redis
```

Comment:
I would get rid of this singleton by going back to the old style of redis instatiation:
```
redis = redis.StrictRedis.from_url(settings.REDIS_URL)
```
And for testing you can use the mock library and it would look like something like:
```
from mock import patch

with patch('formspree.app.redis') as redis_mock:
  ...